lighthouse-treemap/app/src/main.js

```diff
@@ -646,16 +646,14 @@ function renderViewModeButtons(viewModes) {
     if (!viewMode.enabled) viewModeEl.classList.add('view-mode--disabled');
     viewModeEl.id = `view-mode--${viewMode.id}`;
 
-    const inputEl = TreemapUtil.createChildOf(viewModeEl, 'input', 'view-mode__button', {
-      id: `view-mode--${viewMode.id}__label`,
-      type: 'radio',
-      name: 'view-mode',
-      disabled: viewMode.enabled ? undefined : '',
-    });
-
-    const labelEl = TreemapUtil.createChildOf(viewModeEl, 'label', undefined, {
-      for: inputEl.id,
-    });
+    const inputEl = TreemapUtil.createChildOf(viewModeEl, 'input', 'view-mode__button');
+    inputEl.id = `view-mode--${viewMode.id}__label`;
+    inputEl.type = 'radio';
+    inputEl.name = 'view-mode';
+    inputEl.disabled = !viewMode.enabled;
```

Comment:
I never loved the already-long list of string arguments in `createElement`/etc ending with the attrs object, so I'm not sad to see it go, but given that there are other attributes that are unsafe attributes, does it make sense to decentralize the setting of attributes again (vs an allow list inside `createElement`)?

Alternatively: could we poison the unsafe attributes in jsdom so that any new report code (that theoretically has tests that exercises it) will throw in tests? Basically a frontline check for the conformance checks that will likely be catching it a day or two later anyways?

Reply:
> could we poison the unsafe attributes in jsdom

maybe this wouldn't work (easily) because we still need it to work in one place (`safelySetHref`) and it will be annoying to get it working there so the tests can still actually run :)

Reply:
there's also eslint's [`no-restricted-properties `](https://eslint.org/docs/rules/no-restricted-properties) to ban some properties, but I think that would only really work with things like `innerHTML` since we use `.href` liberally in e.g. artifacts, etc. Maybe we could scope it to just the report renderer and have a single `// eslint-disable-line` for `safelySetHref`?

Reply:
i [tried](https://github.com/GoogleChrome/lighthouse/pull/12796#:~:text=%E2%80%A6-,a1b29a7,-Revert%20%22failed%20attempts) to the poisoning. it ends up being particularly gross and borderline impossible without really reaching into jsdom internals. (they have some protection against redefining setters).   this is another thing to revisit after moving the report tests out of jsdom and into the browser.

no-restricted-properties finds all of our .href getters... and it cant' distinguish.


-------

while I'm totally interested in a more substantial check, i don't think we have good tooling to make it straightforward without adding technical debt on its own.

luckily the presubmit in g3 is comprehensive and along with autoroll, we'll have quick feedback there.

Reply:
ah yeah, thanks for trying. Without a type-based analysis like closure, it doesn't look like there's low hanging fruit here. Relying on the presubmit SG